guava/src/com/google/common/base/Optional.java

```diff
@@ -80,14 +83,14 @@
  * @since 10.0
  */
 @GwtCompatible(serializable = true)
-public abstract class Optional<T> implements Serializable {
+public abstract class Optional<T extends Object> implements Serializable {
```

Comment:
These changes are mainly for restrict the instantiation of generic classes to only `@NonNull` types, [see](https://checkerframework.org/manual/#generics) or to default type annotation to be more precise, [see](https://checkerframework.org/manual/#faq-semantics-section). Please let me now if my understanding is at fault. Thanks!

Reply:
This makes sense, thanks.

Reply:
In stub file shared by you, developers have added the following note regarding usage of `extends Object`
```
// NOTE: "extends Object" means there's no explicit annotation so none is added.
// Which means that the default interpretation is used. Since the default
// changed between checker framework versions, "extends Object" translates to:
// - "extends @NonNull Object" in v2.0
// - "extends @Nullable Object" in v2.1+
// TODO: Remove "extends Object".
```
I think I should revert these changes and further should use the fully annotated versions like `extends @NonNull Object` or `extends @Nullable Object` instead of relying on the default interpretation. Let me know how should I proceed. Thanks!

Reply:
Initially I was read through the stub and developed a rationale for including `extends Object`. But later on when I carefully examined the stub file I noticed these comments, which made me question whether or not to include this.
Should I remove these (as developers wanted) or keep these changes?
Thank you.

Reply:
I think smaller diffs are desirable, so I wouldn't add `extends Object` except where necessary.